hols/oci-basic-setup/deploy.sh

```diff
@@ -0,0 +1,145 @@
+#
+
+# Use this to specify the JDK Installer version in tar_gz format
+JDK_TAR_GZ_INSTALLER="https://download.oracle.com/java/21/latest/jdk-21_linux-x64_bin.tar.gz"
+
+HELIDON_MP_APP_ZIP=oci-mp-server.zip
+DEFAULT_PROJECT_PATH=~/oci-mp
+SCRIPT_DIR=$(dirname "$0")
+# shellcheck disable=SC1091
+source "${SCRIPT_DIR}"/get_common.sh
+
+# Main routine
+if [ -z "${1}" ]; then
+    read -r -p "Enter the Helidon MP project's root directory (default: ${DEFAULT_PROJECT_PATH}): " PROJECT_PATH
+    # Use eval to expand ~ if it is part of the input
+    PROJECT_PATH=$(eval echo -n "${PROJECT_PATH:-${DEFAULT_PROJECT_PATH}}")
+    echo "$PROJECT_PATH"
+else
+     PROJECT_PATH=${1}
+fi
+if [ ! -d "${PROJECT_PATH}" ]; then
+    echo "Error: \"${PROJECT_PATH}\" is not a valid directory"
+    exit 1
+fi
+CURRENT_DIR=$(pwd)
+SERVER_BIN_DIR="${PROJECT_PATH}/server/target"
+cd "${SERVER_BIN_DIR}" || exit 1
+
+# Assemble the application zip
+zip -r "${CURRENT_DIR}/${HELIDON_MP_APP_ZIP}" libs oci-mp-server.jar
+cd "${CURRENT_DIR}" || exit 1
+
+# Generate private key file that will be use to ssh or scp to the instance
+"${SCRIPT_DIR}"/get.sh create_ssh_private_key
+# Get instance public IP
+PUBLIC_IP=$("${SCRIPT_DIR}"/get.sh public_ip)
+# Upload the application zip
+scp -o StrictHostKeyChecking=accept-new -i private.key oci-mp-server.zip opc@"${PUBLIC_IP}":/home/opc
+
+# Download & install jdk and run app
+ssh -i private.key opc@"${PUBLIC_IP}" "bash -s ${HELIDON_MP_APP_ZIP} ${JDK_TAR_GZ_INSTALLER}" << 'EOF'
+HELIDON_MP_APP_ZIP=${1}
+JDK_TAR_GZ_INSTALLER=${2}
+
+# Create application and log directory
+export APP_DIR=~/oci-mp/app
+export LOG_DIR=~/oci-mp/log
+mkdir -p "${APP_DIR}"
+mkdir -p "${LOG_DIR}"
+
+# Unzip the application binary
+unzip -o "${HELIDON_MP_APP_ZIP}" -d "${APP_DIR}"
+rm "${HELIDON_MP_APP_ZIP}"
+
+#  Download and extract JDK
+JDK_TAR_GZ_INSTALLER_BASE=$(basename ${JDK_TAR_GZ_INSTALLER})
+if ! ls "${JDK_TAR_GZ_INSTALLER_BASE}" ; then
+    rm -f "*jdk*.tar.gz"
+    rm -rf jdk*
+    # Download JDK
+    curl -O "${JDK_TAR_GZ_INSTALLER}" && echo "JDK downloaded successfully"
+    tar xzf ${JDK_TAR_GZ_INSTALLER_BASE} && echo "JDK installed successfully"
+fi
+
+# Create Service file
+export JAVA_BIN=$(ls -d "$(pwd)"/jdk*/)bin
+cat << EOF_INNER > helidon-app.service.new
+[Unit]
+Description=Helidon OCI-MP application service
+After=syslog.target network.target
+
+[Service]
+User=opc
+Type=simple
+WorkingDirectory=/home/opc
+ExecStart=/bin/bash -c '${JAVA_BIN}/java -jar ${APP_DIR}/oci-mp-server.jar &> ${LOG_DIR}/oci-mp-server.log'
+
+[Install]
+WantedBy=multi-user.target
+EOF_INNER
+
+# Set appropriate SELinux Context for helidon-app.service if SELinux is in enforcing mode
+SELINUX_ENFORCE_STATUS=$(getenforce)
+if [ "${SELINUX_ENFORCE_STATUS}" == "Enforcing" ]; then
+    echo "Setting context for systemd service file"
+    chcon system_u:object_r:systemd_unit_file_t:s0 helidon-app.service.new
+fi
+
+# Start the Helidon application service
```

Comment:
Reminder, somewhere we need to explain what this is intending to accomplish -- since the application is being deployed as a service, we expect that the user wants it to run whenever the instance starts. These next steps set the application up as a service that will automatically be restarted whenever the instance restarts.

Reply:
Updated